File: trunk/resources/scop_1.5.1/python/examples/status.py
#!/usr/bin/python
# status.py - DMI - 8-11-03

import scop

sock = scop.scop_open("localhost", "status")
v = scop.scop_list(sock)
print len(v), "clients connected."
for tuple in v:
	name, interest, src_hint = tuple
	print "Client connection <" + name + "> listening to <" + interest + \
		">, source hint <" + src_hint + ">"
scop.scop_close(sock)

File: trunk/resources/scop_1.5.1/java/examples/Status.java
// Status.java - DMI - 24-12-02

/* Usage: java Status */

import java.io.*;
import java.util.*;

public class Status
{
	public static void main(String[] args)
	{
		SCOP scop;
		Vector v;
		SCOP.ListNode ln;

		scop = new SCOP("localhost", "Status");
		v = scop.list();
		System.out.println(v.size() + " clients connected.");
		for(int i = 0; i < v.size(); i++)
		{
			ln = (SCOP.ListNode)v.elementAt(i);
			System.out.print("Client connection <" + ln.name);
			System.out.print("> listening to <" + ln.interest);
			System.out.print(">, source hint <" + ln.src_hint);
			System.out.println(">");
		}
		scop.close();
	}
}
